src/go/debug-collector/archive/archive.go

```diff
@@ -0,0 +1,58 @@
+package archive
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"io"
+	"os"
+	"path/filepath"
+)
+
+func Create(src string) error {
+	var buf bytes.Buffer
+
+	zr := gzip.NewWriter(&buf)
+	tw := tar.NewWriter(zr)
+
+	filepath.Walk(src, func(file string, fi os.FileInfo, err error) error {
+		header, err := tar.FileInfoHeader(fi, file)
+		if err != nil {
+			return err
+		}
+		header.Name = filepath.ToSlash(file)
+
+		err = tw.WriteHeader(header)
```

Comment:
I'm not sure it's sufficient to have only the name in the header. Mode? Size?

Reply:
https://github.com/percona/percona-toolkit/pull/452/commits/1d6da2fdca6c20f73750aa13d73c9ffc13048964

Reply:
https://github.com/percona/percona-toolkit/pull/452/commits/1d6da2fdca6c20f73750aa13d73c9ffc13048964